src/mds/Server.cc

```diff
@@ -1651,7 +1651,7 @@ void Server::reconnect_tick()
 		      << ", after waiting " << elapse1
 		      << " seconds during MDS startup";
 
-    if (g_conf()->mds_session_blacklist_on_timeout) {
+    if (g_conf()->mds_session_blacklist_on_timeout && !session->is_importing()) {
```

Comment:
I don't think this works

Reply:
Why not? In Server::prepare_force_open_sessions(), the count gets incremented (https://github.com/sidharthanup/ceph/blob/wip-fix-import-recovery/src/mds/Server.cc#L937) . Then this session is persisted in EImportStart. During replay when sessionmap.replay_open_sessions() is called this information is relayed and then Server::reconnect_tick is going to see this and kill it instead of blacklisting.

Reply:
import_count is always 0 after log replay.

Reply:
Ack. Thanks @ukernel. I have updated the code to set a flag on EImportStart replay to avoid blacklisting it. Yet to test it but don't you think this should work?